x-pack/plugins/watcher/public/plugin.ts

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+// import { i18n } from '@kbn/i18n';
+import { CoreSetup, Plugin, CoreStart } from 'kibana/public';
+import { TimeBuckets, MANAGEMENT_BREADCRUMB } from './legacy';
+import { PLUGIN } from '../common/constants';
+import { LICENSE_CHECK_STATE } from '../../licensing/public';
+import { Dependencies } from './types';
+
+export class WatcherUIPlugin implements Plugin<void, void, Dependencies, any> {
+  hasValidLicense: boolean = false;
+  hasRegisteredESManagementSection = false;
+
+  setup(
+    { application, notifications, http, uiSettings, getStartServices }: CoreSetup,
+    { licensing, management, data }: Dependencies
+  ) {
+    licensing.license$.subscribe(license => {
+      const { state } = license.check(PLUGIN.ID, PLUGIN.MINIMUM_LICENSE_REQUIRED);
+      this.hasValidLicense =
+        state === LICENSE_CHECK_STATE.Valid && license.getFeature(PLUGIN.ID).isAvailable;
+
+      if (this.hasValidLicense) {
+        const esSection = management.sections.getSection('elasticsearch');
+        if (esSection) {
+          if (!this.hasRegisteredESManagementSection) {
```

Comment:
All of this logic could be great simplified if an API like this existed, similar to the Application API:

```ts
esSection.registerApp({
  id: 'watcher',
  title: 'Watcher',
  updater$: licensing.license$.pipe(map(license => {
    const { state } = license.check(PLUGIN.ID, PLUGIN.MINIMUM_LICENSE_REQUIRED);
    const hasValidLicense =
      state === LICENSE_CHECK_STATE.Valid && license.getFeature(PLUGIN.ID).isAvailable;
    return {
      status: hasValidLicense ? ManagementAppStatus.accessible : ManagementAppStatus.inaccessible
    };
  })),
  mount(...) { ... },
});
```